server/src/main/java/org/elasticsearch/index/mapper/MappedFieldType.java

```diff
@@ -399,4 +400,20 @@ public TextSearchInfo getTextSearchInfo() {
         KEYWORD,
         NUMERIC
     }
+
+    /**
+     * This method is used to support auto-complete services and implementations
+     * are expected to find terms beginning with the provided string very quickly.
+     * If fields cannot look up matching terms quickly they should return null.  
+     * The returned TermEnum should implement next(), term() and doc_freq() methods
```

Comment:
I kept it in for now because 1) it's cheap to get from the index and 2) Even if we don't use it for popularity sorting we might want to use it later to filter low-frequency items e.g. user tags that occur only once